api/custom_auth/permissions.py

```diff
@@ -17,8 +20,26 @@ def has_object_permission(self, request, view, obj):  # type: ignore[no-untyped-
 
 
 class IsSignupAllowed(AllowAny):
+    message = "Signing-up without a valid invitation is disabled. Please contact your administrator."
+
     def has_permission(self, request: Request, view: View) -> bool:
-        return not settings.PREVENT_SIGNUP
+        if not settings.PREVENT_SIGNUP:
+            return True
+
+        email = request.data.get("email")
+        if email and Invite.objects.filter(email__iexact=email).exists():
```

Comment:
Just checking. Do we have a db index that could favor this query? I'm not entirely sure but the `Invite` model sounds like it can hold many records.